components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissions.kt

```diff
@@ -12,17 +12,22 @@ import mozilla.components.feature.sitepermissions.SitePermissions.Status.NO_DECI
 data class SitePermissions(
     val origin: String,
     val location: Status = NO_DECISION,
+    val doNotAskAgainLocation: Boolean = false,
     val notification: Status = NO_DECISION,
     val microphone: Status = NO_DECISION,
+    val doNotAskAgainMicrophone: Boolean = false,
     val cameraBack: Status = NO_DECISION,
     val cameraFront: Status = NO_DECISION,
+    val doNotAskAgainCamera: Boolean = false,
```

Comment:
I'm confused about the addition of those booleans. Aren't those already covered by the others (e.g. only if the checkbox is checked we need to save anything to the db). Let's chat about that today. :)

Reply:
I think [this video of how Fennec is using the check box](https://drive.google.com/file/d/1176l1-UUZwitgSX1Hv0z8aNmg6ibhdse/view?usp=sharing), can help to clarify.

From what I could saw on Fennec, the checkbox when is checked remember the last user dissension.

For example: If the user checks the checkbox and allows location next time, the user enters again to this site the location will be auto-granted but if the user doesn't check the checkbox a prompt will be always shown either if the user selects allow or deny.

Reply:
I think I got what you meant. The idea is to only store the info when the checkbox is checked, otherwise, not store anything always show the prompt. Right?

Reply:
Yes, that's how I understood it.

Reply:
I updated the PR 👍